OpenImis.ePayment/Escape/Payment/Data/GepgUtility.cs

```diff
@@ -206,245 +187,157 @@ public String CreateGePGCancelPaymentRequest(IConfiguration Configuration, int P
+            var sb = new StringBuilder();
 
-            try
+            using (var xw = XmlWriter.Create(sb, settings))
             {
-
-
-                ns = new XmlSerializerNamespaces();
-                ns.Add("", "");
-                settings = new XmlWriterSettings();
-
-                //settings.Indent = true;
-                StringBuilder sb = new StringBuilder();
-                xs = new XmlSerializer(type);
-                xw = XmlWriter.Create(sb, settings);
-
-                xs.Serialize(xw, content, ns);
-                xw.Flush();
-                outString = sb.ToString();
-            }
-            catch (Exception ex)
-            {
-                Console.WriteLine(ex.ToString());
-            }
-            finally
-            {
-                if (xw != null)
+                try
+                {
+                    xs.Serialize(xw, content, ns);
+                    xw.Flush();
+                }
+                catch (Exception ex)
                 {
-                    xw.Close();
+                    Console.WriteLine(ex.ToString());
                 }
+                return sb.ToString();
             }
-
-            return outString;
         }
 
         public string FinaliseSignedAcks(object content, Type type)
         {
             //Gepg gepgBill = new Gepg() { Content = content, gepgSignature = sign };
 
-            XmlSerializer xs = null;
-            XmlSerializerNamespaces ns = null;
-            XmlWriterSettings settings = null;
-            XmlWriter xw = null;
-            String outString = String.Empty;
+            var xs = new XmlSerializer(type);
+            var ns = new XmlSerializerNamespaces();
+            ns.Add("", "");
+            var settings = new XmlWriterSettings() { Encoding = Encoding.UTF8 };
 
-            try
-            {
-                ns = new XmlSerializerNamespaces();
-                ns.Add("", "");
-                settings = new XmlWriterSettings();
-                settings.Encoding = System.Text.Encoding.UTF8;
-
-                //settings.Indent = true;
-                MemoryStream sb = new MemoryStream();
-                xs = new XmlSerializer(type);
-                xw = XmlWriter.Create(sb, settings);
-
-                xs.Serialize(xw, content, ns);
-                xw.Flush();
-                outString = Encoding.UTF8.GetString(sb.ToArray());
-            }
-            catch (Exception ex)
-            {
-                Console.WriteLine(ex.ToString());
-            }
-            finally
+            using (var sb = new MemoryStream())
+            using (var xw = XmlWriter.Create(sb, settings))
             {
-                if (xw != null)
+                try
                 {
-                    xw.Close();
+                    xs.Serialize(xw, content, ns);
+                    xw.Flush();
+                }
+                catch (Exception ex)
+                {
+                    Console.WriteLine(ex.ToString());
```

Comment:
Instead of logging the error in Console, can we log it to Log4Net or somewhere else to review the exception later?

Reply:
The same as above, i didn't want to modify the component to allow logging